src/librustc_middle/ty/context.rs

```diff
@@ -1207,6 +1208,14 @@ impl<'tcx> TyCtxt<'tcx> {
         if cnum == LOCAL_CRATE { false } else { self.cstore.crate_is_private_dep_untracked(cnum) }
     }
 
+    pub fn crate_extern_paths(&self, cnum: CrateNum) -> Vec<PathBuf> {
```

Comment:
This should be a query: you can declare it be copying this example and updating the signature and description https://github.com/rust-lang/rust/blob/7adbc0dfeffcb6c2ae489cc97d28f76612c7cdf3/src/librustc_middle/query/mod.rs#L1035-L1038